leaf-ui/rappid-extensions/ElementInspector.js

```diff
@@ -316,20 +334,20 @@ var ElementInspector = Backbone.View.extend({
     /**
      * Initializes components to display user defined functions
      */
-    renderUserDefined: function(cell){
+    renderUserDefined: function(){
+        
         this.$('#markedValue').hide();
         $(".function-type").val('UD');
 
         // Load the user defined constraints
-        var len = this.intention.getNumOfFuncSegements();
-        var funcSegments = this.intention.dynamicFunction.getFuncSegmentIterable();
-
+        var len = this.model.get('intention').getFuncSegments().length;
+        var funcSegments = this.model.get('intention').getFuncSegments();
 
         for (var i = 0; i < len; i++) {
 
             // set the intial values
-            $(".user-sat-value").last().val(satisfactionValuesDict[funcSegments[i].funcX].name);
-            $(".user-function-type").last().val(funcSegments[i].funcType);
+            $(".user-sat-value").last().val(satisfactionValuesDict[funcSegments[i].get('refEvidencePair')].name);
```

Comment:
You could use an enhanced for loop here aka (syntax might be slightly off but this is the idea)
```
for (let funcSegment in funcSegments){
$(".user-sat-value").last().val(satisfactionValuesDict[funcSegment.get('refEvidencePair')].name);
etc etc etc
}
```
That way you don't have to then use i to access the funcSegment